### ISSUE TITLE
Regression in 1.1.194 involving mapping-values w/ recursive types

### ISSUE THREAD

**Issue by rsokl** [microsoft/pyright#2999]
`Mapping[Any, <Recursive-Type>]` is not checked accurately; it permits false negatives, as demonstrated below. It appears that this is a regression that was introduced in pyright 1.1.194, and it persists up through 1.1.218.

```python
from typing import Union, Mapping, Any

A = Union[str, int]
B = Union[A, Mapping[Any, "B"]]


class M:
    pass


def g(x: B):
    return


# (pyright 1.1.193) error: Argument of type "dict[Any, M]" cannot be assigned to parameter "x" of type "B"
# (pyright 1.1.[194-218]) no error
g({1: M()})
```

**Comment by erictraut** [comment]
Thanks for the bug report. This will be fixed in the next release.

**Comment by erictraut** [comment]
This is addressed in pyright 1.1.219, which I just published. It will also be in the next release of pylance.
